Feature/pt 3/state management and cart page

flutterVlad opened the pull request:
https://github.com/flutterVlad/food_app/assets/137285738/0e682112-570e-436b-bb23-c68cd33d6045

flutterVlad committed:
[PT-3] feature/PT-3/state-management-and-cart-page

- Added Cart Page
- Added transition from cart page to product detail page
- Added number of items in cart over the cart icon in bottom navigation bar

flutterVlad committed:
[PT-3] feature/PT-3/state-management-and-cart-page

- Added Cart Page
- Added transition from cart page to product detail page
- Added number of items in cart over the cart icon in bottom navigation bar

AliaksandrHuzich commented on features/cart/lib/src/bloc/cart/bloc.dart:
```diff
@@ -0,0 +1,43 @@
+  }
+
+  void _addProduct(CartEvent event, Emitter emit) {
+    state.productList[event.model] = state.productList.containsKey(event.model)
```
List<T> updatedItems = List.of(state.productList);
updated[0] = item;
...
emit(state.copyWith(productsList: updatedList));

AliaksandrHuzich commented on features/cart/lib/src/bloc/cart/bloc.dart:
```diff
@@ -0,0 +1,43 @@
+    emit(
+      LoadedState(
+        countCartProducts: state.countCartProducts + 1,
+        productList: state.productList,
```
naming

AliaksandrHuzich commented on features/cart/lib/src/bloc/cart/event.dart:
```diff
@@ -0,0 +1,19 @@
+part of 'bloc.dart';
+
+abstract class CartEvent {
+  late final ProductModel model;
```
remove

AliaksandrHuzich commented on features/cart/lib/src/bloc/cart/state.dart:
```diff
@@ -0,0 +1,33 @@
+part of 'bloc.dart';
+
+abstract class CartState {
+  final int countCartProducts = 0;
```
same

AliaksandrHuzich commented on features/cart/lib/src/ui/cart_screen.dart:
```diff
@@ -1,11 +1,92 @@
+                ),
+                Container(
+                  margin:
+                      const EdgeInsets.symmetric(vertical: 15, horizontal: 25),
```
,

AliaksandrHuzich commented on features/cart/lib/src/ui/cart_screen.dart:
```diff
@@ -1,11 +1,92 @@
 
   @override
   Widget build(BuildContext context) {
-    return const Placeholder();
```
final CartBloc cartBloc =  BlocProvider.of<ThemeBloc>(context);

AliaksandrHuzich replied:
final Theme theme = Theme.of(context);

AliaksandrHuzich commented on features/cart/lib/src/widgets/list_tile_element.dart:
```diff
@@ -0,0 +1,87 @@
+class ListTileElement extends StatelessWidget {
+  final ProductModel model;
+  final int quantity;
+
```
final VoidCallback onTap;

AliaksandrHuzich commented on features/cart/lib/src/widgets/list_tile_element.dart:
```diff
@@ -0,0 +1,87 @@
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: () {
```
onTap: onTap,

AliaksandrHuzich replied:
routing as bloc event

AliaksandrHuzich reviewed (approved):
files naming

flutterVlad committed:
[PT-3] feature/PT-3/state-management-and-cart-page

- Added navigation as bloc event
- Fixed bloc files naming
- Fixed CartBloc state
